cou/utils/juju_utils.py

```diff
@@ -259,27 +282,18 @@ async def get_status(self) -> FullStatus:
         model = await self._get_model()
         return await model.get_status()
 
-    @retry(no_retry_exceptions=(ActionFailed,))
-    async def _get_action_result(self, action: Action, raise_on_failure: bool) -> Action:
-        """Get results from action.
+    async def list_applications(self) -> dict[str, Application]:
+        """Get all applications related with COU deployed in model.
 
-        :param action: Action object
-        :type: Action
-        :param raise_on_failure: Raise ActionFailed exception on failure, defaults to False
-        :type raise_on_failure: bool
-        :return: action results
-        :rtype: Action
-        :raises ActionFailed: When the application status is in error (it's not 'completed').
+        :return: Map object with app name as key and Application as value
+        :rtype: dict[str, Application]
         """
-        result = await action.wait()
-        if raise_on_failure:
-            model = await self._get_model()
-            status = await model.get_action_status(uuid_or_prefix=action.entity_id)
-            if status != "completed":
-                output = await model.get_action_output(action.entity_id)
-                raise ActionFailed(action, output=output)
-
-        return result
+        model = await self._get_model()
+        return {
+            name: app
+            for name, app in model.applications.items()
+            if is_charm_supported(app.charm_name)
+        }
```

Comment:
We are not using the object `Application` so far, why to not generate the list directly?
```suggestion
        return [
            name
            for name, app in model.applications.items()
            if is_charm_supported(app.charm_name)
        ]
```